inst/shiny/modules/tlg/tlg_option_table.R

```diff
@@ -0,0 +1,158 @@
+    refresh_reactable <- reactiveVal(0)
+
+    #' Shows the editing table
+    observeEvent(input$open_table, {
+      .tlg_option_table_popup()
+    })
+
+    #' Renders the editing table
+    output$table <- renderReactable({
+      #' Create edit widgets using `reactable.extras` based on provided definitions
+      edit_widgets <- imap(opt_def$cols, \(def, colname) {
+        colDef(
+          cell = switch(
+            def$type,
+            text = text_extra(id = session$ns(colname)),
+            select = dropdown_extra(
+              id = session$ns(colname),
+              choices = {
+                if (isTRUE(def$choices == ".colnames")) {
+                  names(data())
+                } else if (length(def$choices) == 1 && grepl("^\\$", def$choices)) {
+                  unique(data()[, sub("^\\$", "", def$choices)])
+                } else {
+                  def$choices
+                }
+              },
+              class = "dropdown-extra"
+            ),
+            stop("Unsupported extra")
+          ),
+          name = def$label
+        )
+      })
+
+      reactable(
+        edits_table(),
+        striped = TRUE,
+        bordered = TRUE,
+        highlight = TRUE,
+        columns = edit_widgets,
+        selection = "multiple",
+        theme = reactableTheme(
+          rowSelectedStyle = list(backgroundColor = "#eee", boxShadow = "inset 2px 0 0 0 #ffa62d")
+        )
+      )
+    }) %>%
+      shiny::bindEvent(list(output_table(), refresh_reactable()))
+
+    #' Attach observers that listen to any changes provided by the user.
+    observe({
+      req(output_table())
+      purrr::walk(names(opt_def$cols), \(colname) {
+        observeEvent(input[[colname]], {
+          edit <- input[[colname]]
+          edited_vars <- edits_table()
+          edited_vars[edit$row, edit$column] <- edit$value
+          edits_table(edited_vars)
+        })
+      })
+    })
+
+    #' Add a new row to the edit table
+    observeEvent(input$add_row, {
+      edits_table() %>%
+        add_row() %>%
+        edits_table()
+
+      shinyjs::runjs("memory = {};") # needed to properly reset reactable.extras widgets
+      refresh_reactable(refresh_reactable() + 1)
+    })
+
+    #' Remove a row from the table
+    observeEvent(input$remove_row, {
+      selected <- getReactableState("table", "selected")
+      req(selected)
+      edits_table()[-selected, ] %>%
+        edits_table()
+      shinyjs::runjs("memory = {};") # needed to properly reset reactable.extras widgets
+      refresh_reactable(refresh_reactable() + 1)
+    })
+
+    #' Confirm changes, apply the changes to output table.
+    observeEvent(input$confirm_changes, {
+      removeModal()
+      output_table(edits_table())
+    })
+
+    #' Cancel changes, close the modal, reset values.
+    observeEvent(input$cancel, {
+      removeModal()
+      shinyjs::runjs("memory = {};") # needed to properly reset reactable.extras widgets
```

Comment:
This line `shinyjs::runjs("memory = {};")` is repeated multiple times. Consider encapsulating this in a helper function to avoid repetition:
```
reset_reactable_memory <- function() {
  shinyjs::runjs("memory = {};")
}
```